use-cases/client-server-msgs-perf/svmsg_file_server.c

```diff
@@ -124,8 +131,13 @@ typedef struct {
  *   system will use next Client_info[] slot, without trying to reuse
  *   the slot(s) freed by clients who exited previously.
  */
-static unsigned int   NumActiveClientsHWM = 0;
-static unsigned int   NumActiveClients    = 0;
+#ifdef __cplusplus
+static std::atomic<int> NumActiveClientsHWM{0};
+static std::atomic<int> NumActiveClients{0};
+#else   // __cplusplus
+static _Atomic int NumActiveClientsHWM = 0;
+static _Atomic int NumActiveClients    = 0;
```

Comment:
The main change needed to synchronize between threads is the conversion of these counter variables to atomic.

The rest seemed to fall in-place fairly (& surprisingly) easily.

Please do cross-check the logic changes around these fields, to make sure I am not making any basic error which will introduce race conditions.

Reply:
Looks good!